### PR TITLE
ethereum workitem

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
@mswiderski couple of workitems to interact with the Ethereum blockchain platform

**Comment by tsurdilo** [comment]
@mswiderski added 3 more workitems to interact and create smart contracts. This is good now for a demo

**Comment by tsurdilo** [comment]
updated all workitems after doing testing vs a private blockchain network. this is still ongoing until workitems are used in a demo

**Comment by tsurdilo** [comment]
after testing in workbench realizing that all of the workitems need to have scope - provided on the jbpm/kie dependencies, otherwise users will have to explicitly define them for each project the workitem handler jars are used. updating for ethereum workitems here but another pr is incoming that changes the depends scopes for all other workitems as well.

**Comment by tsurdilo** [comment]
Jenkins retest please

File: ethereum-workitem/src/test/java/org/jbpm/process/workitem/ethereum/test/PrivateNetworkTest.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                                                 wallet);
+            Credentials credentials = auth.getCredentials();
+
+            //showMyEtherBalance(credentials, web3j);
+            //transactExistingContract(credentials, web3j);
+            //contractSpecificEventListener(web3j);
+            //showCompilers(web3j);
+            //deployContract(credentials, web3j);
+            //queryExistingContract(credentials, web3j);
+            //contractEventListen(web3j);
```
Comment (MarianMacik) [inline ethereum-workitem/src/test/java/org/jbpm/process/workitem/ethereum/test/PrivateNetworkTest.java]: What about this commented code? Shouldn't we delete it?

File: ethereum-workitem/src/test/java/org/jbpm/process/workitem/ethereum/test/PrivateNetworkTest.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+        Transaction transaction = Transaction.createEthCallTransaction(credentials.getAddress(),
+                                                                       "0xa309c58c1F1f582bBaBad952BbfFE40c8D88771e",
+                                                                       EthereumUtils.getEncodedFunction(function));
+
+        EthCall response = web3j.ethCall(
+                transaction,
+                DefaultBlockParameterName.LATEST).sendAsync().get();
+
+        //.sendAsync().get();
```
Comment (MarianMacik) [inline ethereum-workitem/src/test/java/org/jbpm/process/workitem/ethereum/test/PrivateNetworkTest.java]: Commented code.

**Comment by tsurdilo** [comment]
@MarianMacik sorry yes what i wrote was confusing. since these workitems are not built as part of jBPM they need this scope-provided as we expect the container to provide their dependency at runtime.

**Pull request merged**